src/odemis/driver/semnidaq.py

```diff
@@ -1863,6 +1887,253 @@ def _downsample_pixels(cls, data: numpy.array,
             numpy.add.reduce(buffer_osr, axis=2, out=subdata)
 
 
+class ActiveTTLManager:
+    """
+    Manage the "slow" TTL signals (digital output) to indicate the active state of a detector or
+    of the scanner.
+    Typically, it is set to active just before acquisition, and reset to inactive once the acquisition
+    is done. It supports to override the status of each TTL signal independently, via VAs.
+    It also supports a (different) delay before activating
+    """
+    def __init__(self,
+                 sem: AnalogSEM,
+                 parent: model.HwComponent,
+                 ttls: Dict[int, Tuple[bool, bool, Optional[str]]],
+                 activation_delay: float = 0,
+                 on_state_change_cb: Optional[Callable[[Optional[bool]], None]] = None,
+                 ):
+        self._sem = sem
+        self._parent = parent
+        self._on_state_change_cb = on_state_change_cb
+
+        if not 0 <= activation_delay < 1000:
+            raise ValueError("activation_delay %g s is not between 0 and 1000 s" % (activation_delay,))
+        self._activation_delay = activation_delay
+
+        self._ttls = {}  # NI Task -> high_auto (bool), high_enabled (bool), name (str)
+        self._ttl_setters = []  # To hold the partial VA setters
+        self._ttl_tasks = {}  # NI-DAQmx task to control the given D
+        self._ttl_lock = threading.Lock()  # Acquire to change the hw TTL state
+
+        # configure each channel for output
+        available_do_ports = sem.get_available_do_channels(port_number=0)
```

Comment:
Can you write a comment here, why use the port number as 0?

Reply:
The hardware has multiple channels on each port. port 0 is the main one. It has up to 32 channels, and that's the only port we ever used so far. So for now it's hard-coded.
I've added a comment about it.